src/transformers/pipelines/base.py

```diff
@@ -1009,8 +1009,6 @@ def _ensure_tensor_on_device(self, inputs, device):
         elif isinstance(inputs, tuple):
             return tuple([self._ensure_tensor_on_device(item, device) for item in inputs])
         elif isinstance(inputs, torch.Tensor):
-            if device == torch.device("cpu") and inputs.dtype in {torch.float16, torch.bfloat16}:
```

Comment:
Is this good for you @Narsil? Not sure why we previously forced the dtype to fp32 when on cpu, even if the user specified a `torch_dtype` of fp16 or bf16

Reply:
Probably was not supported for all cpus. Still the case, not all cpu support bf16